src/content_script.js

```diff
@@ -32,50 +80,411 @@ function itemWidthCheck (target) {
+  for (let badge of allBadges) {
+    return badge;
+  }
+}
+
+function closePrompt() {
+  const activePrompt = findActivePrompt();
+  activePrompt.classList.remove("active");
+  document.body.classList.remove("js-fbc-prompt-active");
+  document.querySelector(".fbc-has-badge.js-fbc-prompt-active").classList.remove("js-fbc-prompt-active");
+}
+
+function positionPrompt ( target ) {
+  target = document.querySelector("." + target);
+  const targetPrompt = target.querySelector(".fbc-badge-prompt");
+  const elemRect = target.getBoundingClientRect();
+  if ( (window.innerWidth - elemRect.left) < 350  ) {
+    targetPrompt.classList.add("fbc-badge-prompt-align-right");
+  } else {
+    targetPrompt.classList.remove("fbc-badge-prompt-align-right");
+  }
+}
+
+function elementSizeOffsetXY(smallSwitch) {
+  // [X, Y]
+  if (smallSwitch) {
+    return [12, 5];
+  }
+  return [20, 4];
+}
+
+function getOffsetsAndApplyClass(elemRect, bodyRect, target, htmlBadgeDiv) {
+  if ( !isFixed(target) && htmlBadgeDiv.classList.contains("fbc-badge-fixed") ) {
+    htmlBadgeDiv.classList.remove("fbc-badge-fixed");
+  } else if ( isFixed(target) ) {
+    htmlBadgeDiv.classList.add("fbc-badge-fixed");
+    return {offsetPosX: elemRect.left, offsetPosY: elemRect.top};
+  } else {
+    return {offsetPosX: elemRect.left - bodyRect.left, offsetPosY: elemRect.top - bodyRect.top};
+  }
+}
+
+function checkVisibilityAndAppleClass(target, htmlBadgeDiv) {
+  const htmlBadgeDivHasDisabledClass = htmlBadgeDiv.classList.contains("fbc-badge-disabled");
+  const targetIsNull = (target === null);
+
+  if ( targetIsNull && !htmlBadgeDivHasDisabledClass ) {
+    // Element no longer exists and its badge needs to be hidden
+    htmlBadgeDiv.classList.add("fbc-badge-disabled");
+    return;
+  }
+
+  const parentIsHidden = (target.offsetParent === null);
+
+  if ( parentIsHidden && !htmlBadgeDivHasDisabledClass ) {
+    // Parent isnt visible and its badge needs to be hidden
+    htmlBadgeDiv.classList.add("fbc-badge-disabled");
+  }
+
+  if ( !parentIsHidden && !targetIsNull && htmlBadgeDivHasDisabledClass ||  !parentIsHidden && htmlBadgeDivHasDisabledClass ) {
+    htmlBadgeDiv.classList.remove("fbc-badge-disabled");
+  }
+
+}
+
+function calcZindex(target) {
+  const targetParents = [];
+
+  while (target) {
+    targetParents.unshift(target);
+    target = target.parentNode;
+    const parentZindex = window.getComputedStyle(target).getPropertyValue("z-index");
+    if ( parentZindex !== "auto" ) {
+      return parseInt(parentZindex, 10) + 1;
+    }
+  }
+  return 0;
+}
+
+function positionFacebookBadge (target, badgeClassUId, targetWidth, smallSwitch) {
+  // Check for Badge element and select it
+  if (!badgeClassUId) {
+    badgeClassUId = "js-" + target;
+  }
+
+  const htmlBadgeDiv = document.querySelector("." + badgeClassUId);
+
+  // Confirm target element is defined
+  if (!target || !(typeof target === "object")) {
+    target = document.querySelector("." + target);
```

Comment:
This changes `target` variable type from a string to an element? Let's keep the type the same. So maybe the parameter is `targetClass` or `targetUId` and then this is `const targetElement`